### DISCUSSION TITLE
How to deep sort?

### DISCUSSION THREAD

**Discussion by dcheglakov** [directus/directus#17348] [Archive]
Hi, I'm trying to deep sort and it looks like I'm doing something wrong.

I've `m2m` relationship `categories` and `articles` and when I try to sort articles it works only with `id`

```bash
/items/categories?deep[articles][_sort]=id&deep[articles][_limit]=4&fields=articles.articles_id.id,articles.articles_id.date_created
```

<img width="1317" alt="image" src="https://user-images.githubusercontent.com/7033855/215563658-2b79f2fd-801c-4d8b-a0e9-f5c6e2288af5.png">

But when I sort by `date_created` I'm recieving an error

```bash
/items/categories?deep[articles][_sort]=-date_created&deep[articles][_limit]=4&fields=articles.articles_id.id,articles.articles_id.date_created
```

<img width="1316" alt="image" src="https://user-images.githubusercontent.com/7033855/215564426-7a83ce92-b573-4564-a712-06f77e827ba7.png">

- **Directus 9.22.1**
- **PostgreSQL 13.7**
- **Node 18.12.1**

**Comment by br41nslug** [comment]
It looks like the field you're trying to sort by is the relational field `articles_id.date_created` instead of `date_created`.
`deep[articles][_sort]=-articles_id.date_created` might do the trick

**Reply by dcheglakov** [reply]
@br41nslug thanks a lot. It worked.
